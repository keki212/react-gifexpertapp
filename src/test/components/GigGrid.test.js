import React from 'react'
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas del componente <GigGrid />', () => {
    const category = 'One Pounch';
    
    
    test('Debe mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data : [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar items cuando se carga las imagenes', () => {
        const gifs = [
            {
                id: 'ABC',
                url: 'https://localhost/cualquiercosa/cosa.jpg',
                title: 'Cualquier cosa',
                username: 'La cosa'
            },
            {
                id: 'ABD',
                url: 'https://localhost/cualquiercosa/cosa.jpg',
                title: 'Cualquier cosa',
                username: 'La cosa'
            },
            {
                id: 'ABE',
                url: 'https://localhost/cualquiercosa/cosa.jpg',
                title: 'Cualquier cosa',
                username: 'La cosa'
            }
        ]
        useFetchGifs.mockReturnValue({
            data : gifs,
            loading: false
        })
        
        const wrapper = shallow(<GifGrid category={category} />);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
    
})
