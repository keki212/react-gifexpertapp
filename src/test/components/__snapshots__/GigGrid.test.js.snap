// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GigGrid /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInLeft"
  >
    One Pounch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando.....
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas del componente <GigGrid /> Debe mostrar items cuando se carga las imagenes 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInLeft"
  >
    One Pounch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa/cosa.jpg"
      username="La cosa"
    />
    <GifGridItem
      id="ABD"
      key="ABD"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa/cosa.jpg"
      username="La cosa"
    />
    <GifGridItem
      id="ABE"
      key="ABE"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa/cosa.jpg"
      username="La cosa"
    />
  </div>
</Fragment>
`;
